### ISSUE TITLE
Remove database config from dev-vhost

### ISSUE THREAD

**Issue by luboskmetko** [xfiveco/generator-chisel#93]
If I'm not wrong DB config could be removed from dev-vhost.conf as it's now done in wp-config-local.php

**Comment by jakub300** [comment]
Somebody may use global config (the same host, username & password) for all local projects so we may want to support vhost as alternative to local config.

**Comment by luboskmetko** [comment]
Make sense, but in such case wouldn't be better to exclude those from virtual host setting to make it more obvious?

```
SetEnv DB_HOST localhost
SetEnv DB_USER username
SetEnv DB_PASSWORD password

<VirtualHost *:80>
    DocumentRoot "<%= documentRoot %>"
    ServerName <%= serverName %>
    SetEnv DB_NAME <%= dbName %>
   <Directory "<%= documentRoot %>">
      Require all granted
    </Directory>
</VirtualHost>
```

Btw. would you be able to automatically create a database using mysql client? I think this could come quite handy and if somebody doesn't use vhost they could do whole WP setup up to WP installation from command line.

**Comment by jakub300** [comment]
Those are possible scenarios I can think of:
* Developer types all information (without variables) when running `chisel:wp-config`. In that case adding variables to `<VirtualHost>` is useless.
* Developer uses only variables. 
 * He may set all variables inside `<VirtualHost>` because why not. 
 * He may also set some variables globally and others inside `<VirtualHost>` but we can't assume which where.
* Developer uses variables for some fields
 * Most likely all that he not typed are set globally, but it is theoretically possible that he will store some locally inside `<VirtualHost>`.

Considering variety of scenarios, if current solution is not enough, I think we could add SetEnv only for variables he wrote in answers, put them all in `<VirtualHost>` and let developer remove them manually if he has them set globally.

**Comment by thymikee** [comment]
@luboskmetko Since it's neither a bug nor a pain for the developers, and everyone has it's own beloved workflow, and it's just hard to fit into everyones needs, I'd close this one. 

I think there are just no gains for developers and for us too. Or maybe there are some but I'm not aware of 😕?

Regarding automatically creating db config based on mysql client... Lets leave it for later as it seems too be a big time consumer. 

I would just wait for our WP devs to start submitting the issues with our WP workflow and fix then only. It's pretty important that this is a tool for them after all :)

**Comment by thymikee** [comment]
@luboskmetko sorry! I've misread your message about *creating database*. 

This seems not to be _very_ complicated task, but it would require a lot of testing from us and handling edge-cases like "do we have create access?", "do we have valid user/pass?", "will it break the yeoman process in some way?" etc. Couple of hours straight.

I guess the best shot would be to extract it as a separate task and wait for some free time to implement it. What do you think?

**Comment by luboskmetko** [comment]
@thymikee yes, I meant to create database using http://dev.mysql.com/doc/refman/5.7/en/mysql.html with provided credentials (if that's possible). That's not urgent but would be pretty handy, so going to create a separate issue for it.

Then if developer has a wildcard Apache virtual host + a wildcard DNS set up, we could do full WP installation from command line so they can just run `gulp` command once installation is done. Wildcard vhost and DNS setup is a matter of documentation, so no changes in generator are required. There are some guides online and I've just set it up and it works nice with generator. So I would take care of writing some documentation or a blog post about it.

So I'm going to close this issue as it's not relevant anymore. In regards of sample vhost generation the current setup is ok.
